### PR TITLE
Update kind to 1.24.0 and node to 1.31.0

### PR DESCRIPTION
**Pull request by marcushines**
also updates kube apis to 1.31

### CODE REVIEW

**Comment by DanG100** [comment]
It doesn't look you changed in the kindest/node image anywhere? https://github.com/openconfig/kne/blob/main/deploy/kne/kind-bridge.yaml#L10

**Review by DanG100 — APPROVED** [review]

**Comment by DanG100** [comment]
lgtm assuming you can fix presubmit

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openconfig/kne
 
-go 1.21
+go 1.22.0
```
Comment (bstoll) [inline go.mod:3]: There are Go 1.21 version references in:

cloudbuild/external.pkr.hcl
cloudbuild/internal.pkr.hcl
.github/workflows/go.yml
docs/setup.md
x/wire/forward/Dockerfile

We should use the same go version in all of these (and use 1.23.2?)

**Pull request merged**